src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -34,14 +45,24 @@
 }
 
 // Group
-.pf-l-toolbar__group:not(:last-child) {
+.pf-l-toolbar__group {
+  // adds top margin to all group divs
+  margin-top: var(--pf-l-toolbar__group--MarginTop);
   margin-right: var(--pf-l-toolbar__group--MarginRight);
+
+  .pf-l-toolbar__item {
+    margin-top: 0;
+    margin-right: 0;
+  }
 }
 
 // Item
 .pf-l-toolbar__item {
+  margin-top: var(--pf-l-toolbar__item--MarginTop);
+  margin-right: var(--pf-l-toolbar__item--MarginRight);
+
   // Set margin-right so items will align when wrapped
-  .pf-l-toolbar:not(:last-child) {
+  .pf-l-toolbar__item:not(:last-child) {
```

Comment:
I'm not sure that currently we need to support a toolbar item in a toolbar item... but I could be mistaken :)

Reply:
- removed `.pf-l-toolbar__item:not(:last-child) {margin-right: var(--pf-l-toolbar__item--MarginRight);}`